waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -734,8 +739,7 @@
                            :waiter-cluster cluster-name
                            :waiter/user run-as-user}
                   :name k8s-name
-                  ;; TODO - set namespace via x-waiter-namespace service parameter
-                  :namespace run-as-user}
+                  :namespace (or namespace default-namespace run-as-user)}
```

Comment:
I'd prefer that the `run-as-user` be dropped from the namespace calculation.

Reply:
Removed as we discussed (since the integration tests fail if `default-namespace` is not set).